core/v2/types.go

```diff
@@ -83,6 +82,17 @@ type BlobHeader struct {
 	PaymentMetadata core.PaymentMetadata
 }
 
+type BlobHeaderWithoutPayment struct {
```

Comment:
I think `BlobHeaderWithHashedPayment` might be a clearer name for this struct.

Also consider a brief doc, describing why this can/must exist